pytket/pytket/passes/superpass.py

```diff
@@ -0,0 +1,68 @@
+# Copyright 2019-2023 Cambridge Quantum Computing
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import copy
+from typing import Optional, Callable
+
+from pytket.circuit import Circuit
+from .._tket.passes import BasePass
+
+
+class superpass:
```

Comment:
Since this isn't actually a `BasePass`, the name is a bit misleading; maybe it should be something like `PassSelector`.

If we want to make this actually a pass, we should probably do something analogous to the `RepeatWithMetricPass`. Probably not needed unless we want to use it with other pass combinators. (We may want that in the future though.)

Reply:
I have thought about it this should be a Basepass and I came to the conclusion that it is better to not have that. I have done the rename

Reply:
rename done in [fcc0a3f](https://github.com/CQCL/tket/pull/1050/commits/fcc0a3fd67219e86559b3fc7426ffe7909183d45)